### PR TITLE
Better checkpoint saving and loading

### PR DESCRIPTION
**Pull request by sgsmob**
The checkpoint saves are threaded and zipped (re #115).  The loads are now compatible with disconnected graphs (re #112).

### CODE REVIEW

File: spacegraphcats/catlas.py [hunk 1]
```diff
@@ -54,42 +57,65 @@ def load(self, level):
...
             root.write(f)
             f.write("###\n")
             write_to_gxt(f, graph)
+
+    def save(self, graph, nodes, idx, level):
+        """Write out a partial computation."""
+        if self.ignore:
+            return
+        # run on a separate thread because the files can get large
+        thread = Thread(target=self._save, args=(graph, nodes, idx, level))
```
Comment (domoritz) [inline spacegraphcats/catlas.py]: Are we sure that we don't change the graph or nodes? If we do, we need to make a deep copy or should not use threading.

Comment (sgsmob) [inline spacegraphcats/catlas.py]: You are right that the graph and nodes might change.  Creating a deep copy would be very expensive, so I will just remove the threading.

Comment (sgsmob) [inline spacegraphcats/catlas.py]: completed

File: spacegraphcats/catlas.py [hunk 2]
```diff
@@ -31,66 +48,91 @@ def existing(self):
 
         return list(sorted(files))
 
-    def name(self, level):
+    def cp_name(self, level):
```
Comment (domoritz) [inline spacegraphcats/catlas.py]: rename to `checkpoint_name`

**Pull request merged**